### ISSUE TITLE
Make TextEdit behave like LineEdit when pressing the left/right arrow keys with a selection active

### ISSUE THREAD

**Issue by RoniPerson** [godotengine/godot-proposals#1607]
**Describe the project you are working on:** Every project I work on has the problem.

**Describe the problem or limitation you are having in your project:** When editing code you can do selection with shift key combined with arrow keys. In other editors like VSCode it's so that when you let go the shift key and then press the left or right arrow the cursor jumps to the start or end of the selection and deselects everything. But in godot it only jumps one to the left or right. This often stops my workflow when copying values from the same line.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:** The feature would have the option in the settings to enable the jumping mechanics of VSCode. So who has worked with VSCode before can enable it and stick to his programming flow.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
```
if( key = RIGHT && !modifiers.shift && selection.active && settings.feature_active){
    cursor = selection.end;
} else {
    // default behavior
}

if( key = LEFT && !modifiers.shift && selection.active && settings.feature_active){
    cursor = selection.start;
} else {
    // default behavior
}
```

**If this enhancement will not be used often, can it be worked around with a few lines of script?:** I don't think so.

**Is there a reason why this should be core and not an add-on in the asset library?:** It is to small for an add-on. But it is one of the small things that come together to give the best user experience.

**Comment by Calinou** [comment]
> The feature would have the option in the settings to enable the jumping mechanics of VSCode. So who has worked with VSCode before can enable it and stick to his programming flow.

This can probably be always enabled rather than being an option.

**Comment by EricEzaM** [comment]
This is already how it works in LineEdit. Unfortunately LineEdit and TextEdit duplicate a lot of functionality as they do not share the same underlying code (not sure if they can or not), so there are subtle differences between how they work.

**Comment by Paulb23** [comment]
Fixed by godotengine/godot/pull/51502.
